core/src/consensus.rs

```diff
@@ -569,13 +569,36 @@ impl Tower {
                     return SwitchForkDecision::SameFork;
                 }
 
-                assert!(
-                    !last_vote_ancestors.contains(&switch_slot),
-                    "Should never consider switching to slot ({}), which is ancestors({:?}) of last vote: {}",
-                    switch_slot,
-                    last_vote_ancestors,
-                    last_voted_slot
-                );
+                if last_vote_ancestors.contains(&switch_slot) {
+                    if !self.is_stray_last_vote() {
```

Comment:
Ideally, I wanted to move this into heaviest_subtree_fork_choice, but failed. no investigation done. I'm now out of stamina.